pkg/packagekit/updates.jsx

```diff
@@ -513,116 +513,106 @@ class RestartServices extends React.Component {
     }
 }
 
-class ApplyUpdates extends React.Component {
-    constructor() {
-        super();
-        // actions is a chronological list of { status: PK_STATUS_*, package: "name version" } events
-        // that happen during applying updates
-        this.state = { percentage: 0, timeRemaining: null, actions: [] };
-    }
+const ApplyUpdates = ({ transaction, onCancel, allowCancel }) => {
+    const [percentage, setPercentage] = useState(0);
+    const [timeRemaining, setTimeRemaining] = useState(null);
+    // chronological list of { status: PK_STATUS_*, package: "name version" } events that happen during applying updates
+    const [actions, setActions] = useState([]);
+
+    const updateForProperties = (status, packageId) => {
+        // small timeout to avoid excessive overlaps from the next PackageKit progress signal
+        PK.call(transaction, "org.freedesktop.DBus.Properties", "GetAll", [PK.transactionInterface], { timeout: 500 })
+                .then(([props]) => {
+                    const percent = props.Percentage.v;
+                    let remain = -1;
+                    if ("RemainingTime" in props)
+                        remain = props.RemainingTime.v;
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17517-20220705-092548-cbb7d34f-fedora-36-devel//Coverage/lcov/github-pr.diff.gcov.html)